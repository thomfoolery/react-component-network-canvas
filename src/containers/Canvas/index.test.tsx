import React from "react";
import renderer from "react-test-renderer";
import {
  useDragManagerMock,
  useGraphManagerMock,
  useWorkspaceMock,
  useBridgeMock,
} from "@component/utils/mocks";
import {
  useDragManager,
  useGraphManager,
  useWorkspace,
  useBridge,
} from "@component/hooks";

import {Canvas} from "./index";

jest.mock("@component/hooks", () => ({
  useDragManager: jest.fn(),
  useGraphManager: jest.fn(),
  useWorkspace: jest.fn(),
  useBridge: jest.fn(),
}));

useDragManager.mockImplementation(() => useDragManagerMock());
useGraphManager.mockImplementation(() => useGraphManagerMock());
useWorkspace.mockImplementation(() => useWorkspaceMock());
useBridge.mockImplementation(() => useBridgeMock());

describe("Canvas", () => {
  it("renders correctly", () => {
    const transform = `translate3d(100px, 100px, 0) scale(1)`;
    const tree = renderer.create(<Canvas transform={transform} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
