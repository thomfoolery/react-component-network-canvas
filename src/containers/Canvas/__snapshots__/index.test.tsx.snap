// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas renders correctly 1`] = `
<div
  className="Canvas"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "transform": "translate3d(100px, 100px, 0) scale(1)",
    }
  }
>
  <div
    className="SelectBox"
  />
  <svg
    className="CanvasBackground"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      className="isDraft"
      data-testid="cc77e2e0-598e-4cb0-813c-516666955e42"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <path
        className="Edge"
        id="Edge-cc77e2e0-598e-4cb0-813c-516666955e42"
      />
      <path
        className="EdgeArea"
      />
    </g>
  </svg>
</div>
`;
